assets/src/scripts/monitoring-location/components/hydrograph/selectors/discrete-data.js

```diff
@@ -1,17 +1,39 @@
+import isEqual from 'lodash/isEqual';
+import uniqWith from 'lodash/uniqWith';
 import {DateTime} from 'luxon';
 import {createSelector} from 'reselect';
 
 import config from 'ui/config';
 
-import {getGroundwaterLevels} from 'ml/selectors/hydrograph-data-selector';
+import {getGroundwaterLevels, getPrimaryParameter} from 'ml/selectors/hydrograph-data-selector';
+
+const GW_LEVEL_RADIUS = 7;
+const GW_LEVEL_CLASS = 'gw-level-point';
+
+const APPROVAL_QUALIFIERS = {
+    'A': {label: 'Approved', class: 'approved'},
+    'P': {label: 'Provisional', class: 'provisional'},
+    'R': {label: 'Revised', class: 'revised'}
+};
+
+const getDetailsForApprovalCode = function(qualifiers) {
+    const approvalQualifier = qualifiers.find(qualifier => qualifier in APPROVAL_QUALIFIERS);
+    if (approvalQualifier) {
+        return APPROVAL_QUALIFIERS[approvalQualifier];
+    } else {
+        return APPROVAL_QUALIFIERS['P'];
+    }
+};
 
 /*
  * Returns a selector function that returns the groundwater levels that will be visible
- * on the hydrograpnh
+ * on the hydrograph
  * @return {Function} which returns an {Array} of groundwater level object with properties:
  *      @prop {Float} value
- *      @prop {Array of String} qualifiers
  *      @prop {Number} dateTime
+ *      @prop {Array of String} classes - a class that can be used to style this point
```

Comment:
So this is where the properties needed to draw the groundwater level points are determined. This includes the radius.